src/descriptor/bare.rs

```diff
@@ -157,10 +157,17 @@ impl<Pk: MiniscriptKey> fmt::Debug for Bare<Pk> {
 
 impl<Pk: MiniscriptKey> fmt::Display for Bare<Pk> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        use fmt::Write;
-        let mut wrapped_f = checksum::Formatter::new(f);
-        write!(wrapped_f, "{}", self.ms)?;
-        wrapped_f.write_checksum_if_not_alt()
+        let s = format!("{}", self.ms);
+
+        if f.alternate() {
+            write!(f, "{}", s)?;
+        } else {
+            for ch in Checksummed::new_unchecked(&s) {
+                f.write_char(ch)?;
+            }
+        }
```

Comment:
In 22ee854c61b13d15fb7d0b11daf2a87604f45ba7:

We definitely want to keep the `wrapped_f` stuff. This new code is both longer and now allocates.

Reply:
Yeah I balked at this before I pushed it. Will add it back in.